util/mcf_dict_util.py

```diff
@@ -0,0 +1,537 @@
+    """
+    ret_dict = {}
+    dcid_list = get_dcids_node_list(node_list)
+    master_dcid_list = get_dcids_node_list(master_list)
+    for dcid in dcid_list:
+        if dcid in master_dcid_list:
+            ret_dict[dcid] = True
+        else:
+            ret_dict[dcid] = False
+    return ret_dict
+
+
+# drop dcid list
+def drop_nodes(node_list: list, dcid_list: list) -> list:
+    """Drops the node from list of nodes if it's dcid is present in the dcid list.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        dcid_list: List of dcids to be dropped from the list.
+        
+    Returns:
+        List of OrderedDict or Dict objects representing a node in MCF file after dropping.
+    """
+    ret_list = []
+    for node_dict in node_list:
+        if not get_dcid_node(node_dict) in dcid_list:
+            ret_list.append(node_dict)
+    return ret_list
+
+
+def add_path_exp_mcf_dicts(new_path: str,
+                           existing_dict: Optional[dict] = None,
+                           reopen: bool = False) -> dict:
+    """Opens a set of files specified by `new_path` expression to create a list of nodes for each file.
+
+    Args:
+        new_path: Glob like path to open a set of files and create a list of node for each file.
+        existing_dict: (Optional) dict returned from previous call to add more files to same dict object.
+        reopen: Boolean value to skip opening files already present in the dictionary to prevent overwrites.
+        
+    Returns:
+        A dictionary with filenames as objects and a list of OrderedDict objects where each object 
+            represents a node in the MCF file as value.
+    """
+    if not existing_dict:
+        existing_dict = {}
+
+    new_path = os.path.expanduser(new_path)
+    # if is dir add **/*.mcf
+    if os.path.isdir(new_path):
+        new_path = os.path.join(new_path, '**/*.mcf')
+
+    matching_files = glob.glob(new_path, recursive=True)
+    for cur_file in matching_files:
+        # set of files
+        print(cur_file)
+        if cur_file not in existing_dict or reopen:
+            existing_dict[cur_file] = mcf_file_to_dict_list(cur_file)
+
+    return existing_dict
+
+
+def dict_list_to_mcf_str(node_list: list,
+                         sort_keys: bool = False,
+                         regen_complex_vals: bool = False) -> str:
+    """Creates a string (which could be written to a file) from list of Dict like object representation of nodes.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        sort_keys: Boolean value to sort the properties before appending the node to the string.
+                    NOTE: This keeps Node on the top and all inline comments would be moved around.
+        regen_complex_vals: Boolean value to regenrate the value string of the property from the list 
```

Comment:
Please clarify if setting this to `False` wold drop properties with complex values.